stacks-signer/src/runloop.rs

```diff
@@ -164,45 +178,159 @@ impl<C: Coordinator> RunLoop<C> {
         }
     }
 
-    /// Process the event as both a signer and a coordinator
-    fn process_event(
+    /// Handle the block validate response returned from our prior calls to submit a block for validation
+    fn handle_block_validate_response(&mut self, block_validate_response: BlockValidateResponse) {
+        match block_validate_response {
+            BlockValidateResponse::Ok(block_validate_ok) => {
+                // This is a valid block proposal from the miner. Trigger a signing round for it.
+                let (coordinator_id, _) = calculate_coordinator(&self.signing_round.public_keys);
+                if coordinator_id == self.signing_round.signer_id {
+                    // We are the coordinator. Trigger a signing round for this block
+                    let signature_hash = block_validate_ok.block.header.signature_hash().expect("BUG: Stacks node should never return a validated block with an invalid signature hash");
+                    self.commands.push_back(RunLoopCommand::Sign {
+                        message: signature_hash.0.to_vec(),
+                        is_taproot: false,
+                        merkle_root: None,
+                    });
+                }
+            }
+            BlockValidateResponse::Reject(block_validate_reject) => {
+                warn!(
+                    "Received a block proposal that was rejected by the stacks node: {:?}",
+                    block_validate_reject
+                );
+                // TODO: submit a rejection response to the .signers contract for miners
```

Comment:
Will this be done in this PR?

Reply:
Oh sorry I had actually done this...just hadn't pushed the code LOL

Reply:
Done.